### PR TITLE
Evaluation Methods

### PR DESCRIPTION
**Pull request by aliiae**

### CODE REVIEW

File: terms/evaluationmethods_ae.tex [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+However, often improvement is rather incremental~\citep{Hand2006}, so it might not be evident if it is of any importance.
+To decide whether a change is significant, \emph{statistical significance tests} are encouraged, at least in case of scarce testing data~\citep{Drummond2010}.
+
+Details about general statistical tests can be found in~\citep{Urdan2011}, whereas~\citet{Japkowic2011} provide recommendations on the correct use of such tests for classifier comparison.
+In evaluating machine learning techniques, one needs to be cautious when using standard parametric\index{evaluation methods!statistical significance testing!parametric} tests such as the paired $t$-test~\citep{Urdan2011} because of the assumptions expected to be met, e.g.\ equal variance of obtained metrics~\citep{Japkowic2011}. In contrast, \emph{non-parametric}\index{evaluation methods!statistical significance testing!non-parametric} tests do not impose any restriction on the population distribution which makes them more applicable, although less powerful~\citep{Colquhoun1971}.
+%
+Below is an illustration of applying a non-parametric test.
```
Comment (jp-um) [inline terms/evaluationmethods_ae.tex]: Refer to section (next section)  not using propositions "above" and "below" (these may move in the document)

File: terms/evaluationmethods_ae.tex [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+% \nameref{ch:cross-validation},
+on page~\pageref{ch:cross-validation}, so the following is focussed on evaluation metrics and statistical tests.
+
+\section{Performance Measures}
+\citet{Stapor17} notes that according to the \emph{no free lunch theorem}\index{no free lunch theorem}~\citep{Wolpert1996}, it is unreasonable to expect that one classifier is generally better than another since there will always be problems where one outperforms the other and vice versa.
+Evaluation is also highly task-specific and selecting a performance measure should be done with the end goal in mind~\citep{goodfellow2016deeplearning}. For example, natural language generation models used in image captioning are penalised for ``hallucinating'' extra information, i.e.\ such models are considered worse, but this aspect can be encouraged when applied to poetry generation.
+
+\subsection{Automatic Evaluation Metrics}
+For supervised classification, the simplest measure is \emph{accuracy}\index{confusion matrix!accuracy} which is the ratio of correctly predicted examples, and its reverse metric \emph{error rate}\index{evaluation methods!error rate}, i.e.\ the fraction of errors.
```
Comment (jp-um) [inline terms/evaluationmethods_ae.tex]: This index entry should not be under confusion matrix -- but under evaluation methods !!!  Please fix.

File: terms/evaluationmethods_ae.bib [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    title = {Intrinsic vs. Extrinsic Evaluation Measures for Referring Expression Generation},
+    booktitle = {Proceedings of the 46th Annual Meeting of the Association for Computational Linguistics on Human Language Technologies: Short Papers},
+    year = {2008},
+    location = {Columbus, Ohio},
+    pages = {197--200},
+    numpages = {4},
+    publisher = {Association for Computational Linguistics},
+}
+
+@misc{Dua2017,
```
Comment (jp-um) [inline terms/evaluationmethods_ae.bib]: what is this a book?  a technical report?  Please make sure you select correct type of publication

Comment (aliiae) [reply terms/evaluationmethods_ae.bib]: If you mean Dua2017, it is the UCI website cited [as per their policy](https://archive.ics.uci.edu/ml/citation_policy.html).

Comment (jp-um) [reply terms/evaluationmethods_ae.bib]: Thanks for the correction, I'll take it as is.

File: terms/evaluationmethods_ae.tex [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+However, often improvement is rather incremental~\citep{Hand2006}, so it might not be evident if it is of any importance.
+To decide whether a change is significant, \emph{statistical significance tests} are encouraged, at least in case of scarce testing data~\citep{Drummond2010}.
+
+Details about general statistical tests can be found in~\citep{Urdan2011}, whereas~\citet{Japkowic2011} provide recommendations on the correct use of such tests for classifier comparison.
+In evaluating machine learning techniques, one needs to be cautious when using standard parametric\index{evaluation methods!statistical significance testing!parametric} tests such as the paired $t$-test~\citep{Urdan2011} because of the assumptions expected to be met, e.g.\ equal variance of obtained metrics~\citep{Japkowic2011}. In contrast, \emph{non-parametric}\index{evaluation methods!statistical significance testing!non-parametric} tests do not impose any restriction on the population distribution which makes them more applicable, although less powerful~\citep{Colquhoun1971}.
+%
+Below is an illustration of applying a non-parametric test.
+
+\subsection{An Illustration of Statistical Significance Testing}
```
[the remark this answers is not in the document]
Comment (aliiae) [reply terms/evaluationmethods_ae.tex]: Done.

File: terms/evaluationmethods_ae.tex (hunk 2 above)
Comment (aliiae) [reply terms/evaluationmethods_ae.tex]: Done. I wanted to reuse the previously added index for accuracy, that's why it was under confusion matrix.

File: terms/evaluationmethods_ae.tex (hunk 1 above)
Comment (aliiae) [reply terms/evaluationmethods_ae.tex]: Done. Replaced with "what follows is", as I do not see much problem with this formulation even if the next section moves to the next page.

**Comment by jp-um** [comment]
Merged, thanks.

**Pull request merged**